exasol_advanced_analytics_framework/testing/mock_query_handler_runner.py

```diff
@@ -0,0 +1,102 @@
+from typing import Callable, TypeVar, Generic, Tuple
+
+from exasol_data_science_utils_python.udf_utils.sql_executor import SQLExecutor
+
+from exasol_advanced_analytics_framework.query_handler.context.scope_query_handler_context import \
+    ScopeQueryHandlerContext
+from exasol_advanced_analytics_framework.query_handler.context.top_level_query_handler_context import \
+    TopLevelQueryHandlerContext
+from exasol_advanced_analytics_framework.query_handler.query.select_query import SelectQueryWithColumnDefinition
+from exasol_advanced_analytics_framework.query_handler.query_handler import QueryHandler
+from exasol_advanced_analytics_framework.query_handler.result import Continue, Finish, Result
+from exasol_advanced_analytics_framework.query_result.mock_query_result import MockQueryResult
+from exasol_advanced_analytics_framework.udf_framework.query_handler_runner_state import QueryHandlerRunnerState
+
+ResultType = TypeVar("ResultType")
+ParameterType = TypeVar("ParameterType")
+
+
+class MockQueryHandlerRunner(Generic[ParameterType, ResultType]):
+
+    def __init__(self,
+                 sql_executor: SQLExecutor,
+                 top_level_query_handler_context: TopLevelQueryHandlerContext,
+                 parameter: ParameterType,
+                 query_handler_factory: Callable[
+                     [ParameterType, ScopeQueryHandlerContext],
+                     QueryHandler[ParameterType, ResultType]]):
+        self._sql_executor = sql_executor
+        query_handler = query_handler_factory(parameter, top_level_query_handler_context)
+        self._state = QueryHandlerRunnerState(
+            top_level_query_handler_context=top_level_query_handler_context,
+            query_handler=query_handler
+        )
+
+    def run(self) -> ResultType:
+        try:
+            result = self._state.query_handler.start()
+            if isinstance(result, Finish):
+                self.handle_finish()
+                return result.result
+            else:
+                while isinstance(result, Continue):
+                    result = self.handle_continue(result)
+                if isinstance(result, Finish):
+                    self.handle_finish()
+                    return result.result
+                else:
+                    raise RuntimeError("Unknown Result")
+        except Exception as e:
+            self.handle_finish()
+            raise RuntimeError(f"Execution of query handler {self._state.query_handler} failed.") from e
+
+    def handle_continue(self, result: Continue):
+        self.release_and_create_new_return_query_query_handler_context()
+        self.cleanup_query_handler_context()
+        for query in result.query_list:
+            self._sql_executor.execute(query.query_string)
+        input_query_result = self.run_input_query(result)
+        result = self._state.query_handler.handle_query_result(input_query_result)
+        return result
```

Comment:
```suggestion
        return self._state.query_handler.handle_query_result(input_query_result)
```

Reply:
I prefer to assign the result of a function first to variable, because this make debugging easier